source/MaterialXCore/Value.h

```diff
@@ -193,6 +194,65 @@ template <class T> class MX_CORE_API TypedValue : public Value
     T _data;
 };
 
+/// The class template for typed subclasses of Value
+class MX_CORE_API AggregateValue : public Value
+{
+  public:
+    AggregateValue(const string& typeName) :
+        _typeName(typeName)
+    {
+    }
+    virtual ~AggregateValue() { }
+
+    /// Create a deep copy of the value.
+    ValuePtr copy() const override
+    {
+        auto result = createAggregateValue(_typeName);
+        for (const auto& val : _data)
+        {
+            result->appendValue(val->copy());
+        }
+        return result;
+    }
+
+    /// Append a member value to the aggregate.
+    void appendValue(ConstValuePtr valuePtr) {
```

Comment:
For consistency, these brackets should use Allman conventions, where each bracket is placed on its own line.  The same applies to other brackets in this changelist, but I'll let this comment stand in for all of them.